src/app/common/services/gateways/gateways.service.ts

```diff
@@ -49,64 +50,91 @@ export class GatewaysService {
     return this.thingsService.addThing(gateway).map(
       resp => {
         const gwID = resp.headers.get('location').replace('/things/', '');
-
-        const ctrlChan: Channel = {
-          name: `${gateway.name}-${this.typeCtrlChan}`,
-          metadata: {
-            type: this.typeCtrlChan,
-          },
-        };
-        this.channelsService.addChannel(ctrlChan).subscribe(
-          respAddCtrl => {
-            const ctrlChanID = respAddCtrl.headers.get('location').replace('/channels/', '');
-
-            const dataChannel: Channel = {
-              name: `${gateway.name}-${this.typeDataChan}`,
+        this.thingsService.getThing(gwID).subscribe(
+          (respGetThing: any) => {
+            gateway.key = respGetThing.key;
+            const ctrlChan: Channel = {
+              name: `${gateway.name}-${this.typeCtrlChan}`,
               metadata: {
-                type: this.typeDataChan,
+                type: this.typeCtrlChan,
               },
             };
-            this.channelsService.addChannel(dataChannel).subscribe(
-              respAddData => {
-                const dataChanID = respAddData.headers.get('location').replace('/channels/', '');
-
-                this.channelsService.connectThing(ctrlChanID, gwID).subscribe(
-                  respConnectCtrl => {
-                    this.channelsService.connectThing(dataChanID, gwID).subscribe(
-                      respConnectData => {
-                        gateway.metadata.ctrlChannelID = ctrlChanID;
-                        gateway.metadata.dataChannelID = dataChanID;
-                        gateway.metadata.gwPassword = uuid();
-                        gateway.id = gwID;
-
-                        this.thingsService.editThing(gateway).subscribe(
-                          () => {
-                            this.notificationsService.success('Gateway successfully created', '');
-
-                            // Send fake location
-                            this.messagesService.sendLocationMock(dataChanID, gwID);
-
-                            // Bootstrap gateway
-                            this.bootstrapService.addConfig(gateway).subscribe();
+            this.channelsService.addChannel(ctrlChan).subscribe(
+              respAddCtrl => {
+                const ctrlChanID = respAddCtrl.headers.get('location').replace('/channels/', '');
```

Comment:
If you use `ctrlChanID` it might not be initialized before you use it, cause you use it outside of the subscription in the line 86.

Reply:
My bad. Thought it was an HTTP req, while it's only a resp parsing. LGTM.